src/t_list.c

```diff
@@ -225,13 +225,14 @@ void pushGenericCommand(client *c, int where) {
         return;
     }
 
+    if (!lobj) {
```

Comment:
It looks like this function was written so that it was "correct" if you passed in a client without any elements to push. Now, this will create an invalid list, since it will be empty. I'm okay with updating this assumption, but we should also update ```(lobj ? listTypeLength(lobj) : 0))```  and ```if (pushed) {``` checks, since they are unnecessary if we change this assumption.

Reply:
These two pieces of code really need to be optimized. I'm not sure that adding `serverAssert(pushed != 0)` is appropriate.

Reply:
There is also a lot of code that determines if (pushed), and it feels like we can use serverAssert instead of pushed.
similar command: lpushxCommand, rpushxCommand

Reply:
Maybe we should just merge pushGenericCommand and pushxGenericCommand? They do the same thing after either creating the key or exiting if it doesn't exist. 

I don't really think an assert is necessary.